### DISCUSSION TITLE
Cask M1 support

### DISCUSSION THREAD

**Discussion by sosterberg** [Homebrew/discussions#721] [Casks] [answered]
Hi,

How do I tell that a cask supports Apple Silicon? For instance, zoom is supposed to have an ARM version, but WhatsApp doesn't; both formulae pages look quite the same with no architecture info. (https://formulae.brew.sh/cask/zoom vs. https://formulae.brew.sh/cask/whatsapp)

Thanks in advance,

**Comment by SMillerDev** [comment, marked as answer]
Since there is no syntax to define specific architectures in casks, you can't. You'll just have to check the code unfortunately (and even then it could be a fat binary that supports both)

**Reply by sosterberg** [reply]
Ok, I see...

But if my architecture isn't supported, will it break or will it fall back on Rosetta 2?

IF it falls back on Rosetta, will an Apple Silicon-supporting future version automagically correct itself?

Thanks again,

**Reply by carlocab** [reply]
Casks just use whatever installer that upstream provides. If the app is not native, then it should run in Rosetta.

Whether it becomes native in the future depends on how upstream decides to distribute their native package. If they distribute a universal installer, then yes. If not, then the cask will need to be updated first to account for the separate M1-native package.
